modules/raveboard/src/app/socketio/socketio.service.ts

```diff
@@ -0,0 +1,34 @@
+import {Injectable} from "@angular/core";
+import {Observable, Subject} from "rxjs";
+import {filter, map} from "rxjs/operators";
+import {ActivationUpdate, SpikeUpdate} from "../model/model";
+import * as io from "socket.io-client";
+
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SocketIOService {
+
+    dataStream: Subject<SpikeUpdate | ActivationUpdate>;
+
+    activations: Observable<ActivationUpdate>;
+    spikes: Observable<SpikeUpdate>;
+
+    constructor() {
+        this.dataStream = new Subject();
+        this.activations = this.dataStream.pipe(filter(data => data.type === 'activation'), map(data => data as ActivationUpdate));
+        this.spikes = this.dataStream.pipe(filter(data => data.type === 'spike'), map( data => data as SpikeUpdate));
+
+        let socket = io.connect("http://localhost:4242");
```

Comment:
The URL should be configurable by environment vars or UI inputs at some point later. Fine for this PR.

Reply:
Done

Reply:
Where do we usually store all configs?
Right now, on the python side, the port (4242) is set in raveboard/ui_context.py
The UI is still hardcoded to 4242 here. Should we have a JSON file that both python and UI can read or something similar? A definition in one single place would be great.

Reply:
Socket.IO URL is now read from `rs-sio-uri` URL parameter.